RepoCleanup/Functions/MigrateXsdSchemasFunction.cs

```diff
@@ -0,0 +1,170 @@
+                }
+
+                List<string> changedFiles = Status(repoFolder);
+
+                if (changedFiles.Count > 0)
+                {
+                    var commitIfChangesCommand = new CommitChangesCommand(repoFolder);
+                    var commitIfChangesCommandHandler = new CommitChangesCommandHandler(new GiteaService());
+                    await commitIfChangesCommandHandler.Handle(commitIfChangesCommand);
+
+                    PushChangesCommand pushChangesCommand = new PushChangesCommand(remotePath, repoFolder, Globals.GiteaToken);
+                    PushChangesCommandHandler.Handle(pushChangesCommand);
+                }
+            }
+
+            logger.WriteLog();
+            await Task.Delay(1);
+
+            return;
+        }
+
+        private static async Task CreateFolderStructure(string repoFolder)
+        {
+            Directory.CreateDirectory($"{repoFolder}\\.altinnstudio");
+            Directory.CreateDirectory($"{repoFolder}\\altinn2");
+            Directory.CreateDirectory($"{repoFolder}\\shared");
+
+            await System.IO.File.WriteAllTextAsync(
+                $"{repoFolder}\\.altinnstudio\\settings.json",
+                "{\n  \"repotype\" : \"datamodels\"\n}");
+
+            await System.IO.File.WriteAllTextAsync(
+                $"{repoFolder}\\shared\\README.md",
+                "# Shared models");
+        }
+
+        private static string CollectMigrationWorkFolder()
+        {
+            Console.WriteLine("This operation requires a folder to which it can clone the datamodels repositories.");
+            string basePath = SharedFunctionSnippets.CollectInput("Provide folder name (should be empty): ");
+
+            if (!Directory.Exists(basePath))
+            {
+                Console.WriteLine("Can't find the specified folder!");
+                return CollectMigrationWorkFolder();
+            }
+
+            return basePath;
+        }
+
+        /// <summary>
+        /// List the GIT status of a repository
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="repository">The name of the repository</param>
+        /// <returns>A list of changed files in the repository</returns>
+        public static List<string> Status(string localRepoPath)
+        {
+            List<string> repoContent = new List<string>();
+            using (var repo = new LibGit2Sharp.Repository(localRepoPath))
+            {
+                Commands.Stage(repo, "*");
+                RepositoryStatus status = repo.RetrieveStatus(new StatusOptions());
+                foreach (StatusEntry item in status)
+                {
+                    repoContent.Add(item.FilePath);
+                }
+            }
+
+            return repoContent;
+        }
+
+        private static async Task<List<string>> CollectOrgInfo()
```

Comment:
I did create a function for this as well, but that was one of the things I forgot to merge.